### DISCUSSION TITLE
Creating Executable Code Process

### DISCUSSION THREAD

**Discussion by AnirudhChaudhary** [Cornutum/tcases#184] [Q&A] [answered]
Hi there. I was wondering if someone could clarify the process of using Tcases for me. 

I run TCases on my API with the command Tcases-api xyz.json .

I get an output of the form xyz-Requests-Test.json. To clarify, this is a test definition file for the xyz file that I ran the tool on. This test definition file describes what should be tested from the original input file.

From here we can create a generator file which combines variables together. Is there a difference between variables and tuples? Do we have to manually create the generator file for TCases by writing out the different definitions and combinations for the tuples/variables?

Assuming that I use the 1-tuple coverage for Tcases, I desire to create JUnit tests for my API. 

I run Tcases-api -J xyz.json and the result I get is a .java file named xyzRequestsTest.java. From here I'm confused about how to create a document that has actually executable code. This xyzRequestsTest.java file seems to descibe the different combinations of items to test but doesn't have code that can be run. 

I noticed that there was a section where we could transform the Input file directly into executable code. I ran a command detailed as : Tcases -x annotations-Transform.xsl annotations-Input.xml and I was able to create a document of executable code that I am looking for. I don't however understand how I can create this .xsl file for my own use because when I tried doing Tcases -x annotations-Transform.xsl Ice-Cream-Input.xsl, the resulting file was not working.

I would really appreciate if someone could clarify this process to me.

Thanks for your time.

**Comment by kerrykimbrough** [comment, marked as answer]
You can learn how to do this [here](https://github.com/Cornutum/tcases/blob/master/tcases-openapi/Running-Api-Test-Cases.md).

Sorry that this was unclear. The path to create this capability took several steps, over an extended period of time. The `tcases-api` command was the first step toward this goal, but the goal was not reached until much later, when the `tcases-api-test` command was complete. Since then, most of the interest is in `tcases-api-test` and there is less interest in `tcases-api`.
